### PR TITLE
#10941 update trial issue

### PR DESCRIPTION
**Pull request by rokonec**
https://github.com/dotnet/core-eng/issues/10941

The code start to be hard to follow, so I refactored it for readability. Performance is not a concern here as related part of code is run only few times a day.

### CODE REVIEW

File: src/DotNet.Status.Web.Tests/DotNet.Status.Web.Tests.csproj [hunk 1]
```diff
@@ -10,6 +10,12 @@
     <None Remove="Files\triage-items-with-diff-cat.expected.json" />
     <None Remove="Files\triage-items-with-invalid.body.txt" />
     <None Remove="Files\triage-items-with-invalid.expected.json" />
+    <None Remove="Files\triage-items-with-multi-builds-intersects.body.txt" />
```
Comment (premun) [inline src/DotNet.Status.Web.Tests/DotNet.Status.Web.Tests.csproj:13]: I think there is something inherently wrong with having a folder named `Files` 😁 something like a Hungarian notation for file system

File: src/DotNet.Status.Web.Tests/Files/triage-items-with-diff-cat.body.txt [hunk 2]
```diff
@@ -1,3 +1,21 @@
+Error message
+```
+We stopped hearing from agent Azure Pipelines 11. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?li
+```
+Builds effected
```
Comment (lukas-lansky) [inline src/DotNet.Status.Web.Tests/Files/triage-items-with-diff-cat.body.txt:5]: Nit: Maybe "affected" would be a better word here given it's not "builds created because of the problem", but "builds where the problem was observed"?

**Review by lukas-lansky — APPROVED** [review]
